source/slang/slang-ir-link.cpp

```diff
@@ -1387,8 +1389,21 @@ LinkedIR linkIR(
     //
     insertGlobalValueSymbols(sharedContext, targetProgram->getExistingIRModuleForLayout());
 
-
     auto context = state->getContext();
+
+    // Combine all of the contents of IRGlobalHashedStringLiterals
+    {
+        StringSlicePool pool;
```

Comment:
I'm curious why a `StringSlicePool` gets involved here, rather than just collecting the list of `IRStringLit`s or similar from the input modules, and then cloning those into the destination. Using the pool seems to introduce an extra intermediate copy between the source(s) and destination.

Reply:
Well.. it was originally like that - but because of the need for a SharedIRBuilder and a module to put them in, and having to copy them into said module, it turned out to be significantly simpler with StringSlicePool.